File: chapter-04-dslcontext/code-java/src/main/java/com/example/service/JooqHealthCheckService.java
package com.example.service;

import lombok.RequiredArgsConstructor;
import org.jooq.DSLContext;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class JooqHealthCheckService {

    // Spring Boot Auto-configuration에 의해 DSLContext가 자동으로 주입됩니다.
    private final DSLContext dsl;

    public Integer checkHealth() {
        // 가장 단순한 쿼리를 날려봅니다. (SELECT 1)
        return dsl.selectOne().fetchOneInto(Integer.class);
    }
}

File: chapter-04-dslcontext/code-kotlin/src/main/kotlin/com/example/service/JooqHealthCheckService.kt
package com.example.service

import org.jooq.DSLContext
import org.springframework.stereotype.Service

@Service
class JooqHealthCheckService(
    // 생성자 주입 방식으로 Spring Boot가 DSLContext 빈을 자동 연결해 줍니다.
    private val dsl: DSLContext
) {
    fun checkHealth(): Int? {
        // 코틀린 환경에서도 직관적인 jOOQ DSL 사용
        return dsl.selectOne().fetchOneInto(Int::class.java)
    }
}
